<div>
Olá, Chau!

Meu nome é Luiz. Fico feliz em revisar seu projeto. Ao longo do texto farei algumas observações sobre melhorias no código e também farei comentários sobre suas percepções sobre o assunto. Estarei aberto a feedbacks e discussões sobre o tema.

**Peço que mantenha e não altere os comentários que eu fizer por aqui para que possamos nos localizar posteriormente, ok?**

Mais uma coisa, vamos utilizar um código de cores para você entender o meu feedback no seu notebook. Funciona assim:

<div class="alert alert-block alert-success">
<b> Comentário do revisor: </b> <a class="tocSkip"></a>

Sucesso. Tudo foi feito corretamente.
</div>

<div class="alert alert-block alert-warning">
<b>Comentário do revisor: </b> <a class="tocSkip"></a>

Alerta não crítico, mas que pode ser corrigido para melhoria geral no seu código/análise.
</div>

<div class="alert alert-block alert-danger">

<b>Comentário do revisor: </b> <a class="tocSkip"></a>
    
Erro que precisa ser arrumado, caso contrário seu projeto **não** será aceito.
</div>

Você pode interagir comigo através dessa célula:
<div class="alert alert-block alert-info">
<b>Resposta do Aluno.</b> <a class="tocSkip"></a>
</div>

### Introdução
1. Importando bibliotecas.
2. Criando caminho para os dados.
3. Leitura dos dados.


In [3]:
import pandas as pd

In [4]:
marketing_events = r"ab_project_marketing_events_us.txt"
events_upd = r"final_ab_events_upd_us.txt"
new_users = r"final_ab_new_users_upd_us.txt"
part = r"final_ab_participants_upd_us.txt"

In [7]:
marketing = pd.read_csv(marketing_events)
events = pd.read_csv(events_upd)
users = pd.read_csv(new_users)
part = pd.read_csv(part)

In [12]:
marketing.head()

,name,regions,start_dt,finish_dt
0,Christmas&New Year Promo,"EU, N.America",2020-12-25,2021-01-03
1,St. Valentine's Day Giveaway,"EU, CIS, APAC, N.America",2020-02-14,2020-02-16
2,St. Patric's Day Promo,"EU, N.America",2020-03-17,2020-03-19
3,Easter Promo,"EU, CIS, APAC, N.America",2020-04-12,2020-04-19
4,4th of July Promo,N.America,2020-07-04,2020-07-11
5,Black Friday Ads Campaign,"EU, CIS, APAC, N.America",2020-11-26,2020-12-01
6,Chinese New Year Promo,APAC,2020-01-25,2020-02-07
7,Labor day (May 1st) Ads Campaign,"EU, CIS, APAC",2020-05-01,2020-05-03
8,International Women's Day Promo,"EU, CIS, APAC",2020-03-08,2020-03-10
9,Victory Day CIS (May 9th) Event,CIS,2020-05-09,2020-05-11


In [9]:
events.head()

,user_id,event_dt,event_name,details
0,E1BDDCE0DAFA2679,2020-12-07 20:22:03,purchase,99.99
1,7B6452F081F49504,2020-12-07 09:22:53,purchase,9.99
2,9CD9F34546DF254C,2020-12-07 12:59:29,purchase,4.99
3,96F27A054B191457,2020-12-07 04:02:40,purchase,4.99
4,1FD7660FDF94CA1F,2020-12-07 10:15:09,purchase,4.99


In [10]:
users.head()

,user_id,first_date,region,device
0,D72A72121175D8BE,2020-12-07,EU,PC
1,F1C668619DFE6E65,2020-12-07,N.America,Android
2,2E1BF1D4C37EA01F,2020-12-07,EU,PC
3,50734A22C0C63768,2020-12-07,EU,iPhone
4,E1BDDCE0DAFA2679,2020-12-07,N.America,iPhone


In [11]:
part.head()

,user_id,group,ab_test
0,D1ABA3E2887B6A73,A,recommender_system_test
1,A7A3664BD6242119,A,recommender_system_test
2,DABC14FDDFADD29E,A,recommender_system_test
3,04988C5DF189632E,A,recommender_system_test
4,4FF2998A348C484F,A,recommender_system_test


# Decomposição do projeto.
1. Avaliar se o novo sistema de recomendação teve uma melhoria de mais de 10% nas taxas do funil(Visualiazação de páginas, adicionar item ao carrinho e finalizar compra)
2. Os dados foram coletados corretamente?
3. Público alvo correto?
4. Grupos A e B equilibrados?
5. Há diferenças significativas entre os grupos nas métricas?
6. O efeito é estatisticamente significativo?


# Plano de trabalho
1. Carregar todos os CSVs
2. Conferir tipos de dados e valores ausentes / duplicidades.
3. Verificar datas.
### Validação do teste A/B.
1. Verificar participantes(Usúarios e Grupos)
2. Tamanho dos grupos.
3. Diferença entre os grupos > 10%
4. 15% dos novos usuários da UE
5. Construção do Funil
6. Teste estatístico
7. Visualizações dos dados.
8. Conclusõoes e recomendações.

<div class="alert alert-block alert-success">
<b> Comentário geral do revisor</b> <a class="tocSkip"></a>

Obrigado por enviar o seu projeto e pelo esforço de chegar até aqui. Aqui você poderia incluir/pensar em análises para responder à hipótese acima. Lembre-se que a hipótese é o ponto principal que gostaríamos de validar, porém é preciso realizar perguntas que devem ser respondidas com às análises durante o projeto.

Seguem alguns pontos como sugestão:

- Você pode analisar a correlação entre o valor de compras/carrinhos e atributos numéricos e/ou categóricos
- Usar técnicas de agrupamento (Kmeans, por exemplo) para agrupar os usuários com base em seu comportamento. Dessa forma, você poderia pensar em estratégias de marketing específicas para cada grupo de usuários
- Realizar uma análise de sobrevivência para responder à seguinte pergunta: a partir de quantas compras um cliente é "fidelizado"? Pense no esforço mínimo que a empresa precisa investir para conquistar um cliente fidelizado.
       
<br>   
    
Qualquer dúvida, pode contar comigo.   
    
<br>  
    
**Até breve!**

</div>